Modules/TRD/src/CheckOnHc2d.cxx

```diff
@@ -0,0 +1,204 @@
+              return result1;
+            } // endif chamber is not masked
+
+            // else{
+            //   nMaskedHC++;
+            //   nMaskedChInASec++;
+            //   if(nMaskedHC >  int(NCHAMBER*2*0.4) ){ // if nMaskedHC is greator than 40% of all chambers
+            //     result1 = Quality::Bad;
+            //     result1.addReason(FlagReasonFactory::Unknown(), "more than 40 percent chambers are masked!");
+            //     return result1;
+            //   }
+            //   if(nMaskedChInASec == NCHAMBERPERSEC){ //if entire sector is masked
+            //     result1 = Quality::Bad;
+            //     result1.addReason(FlagReasonFactory::Unknown(), "some sectors are entirely masked");
+            //     return result1;
+            //   }
+            // }
+
+          } // endif empty half-chamber
+        }   // stack_layer loop ends
+      }     // sector_side loop ends
+
+      result2 = Quality::Good;
+      int nMaskedHC = 0;        // total number of masked half-chamber
+      int nMaskedHChInASec = 1; // total number of masked half-chamber in a sector
+      int currentSector = -1, oldSector = -1;
+      for (int hcid = 0; hcid < MAXCHAMBER * 2; ++hcid) {
+        currentSector = Helper::getSector(hcid / 2);
+        if (TRDHelpers::isHalfChamberMasked(hcid, mChamberStatus)) {
+          ILOG(Debug, Trace) << "Masked half Chamber id =" << hcid << ENDM;
+          nMaskedHC++;
+          if (nMaskedHC > (NCHAMBER * 2 * 0.4)) { // if nMaskedHC is greator than 40% of all chambers
+            result2 = Quality::Bad;
+            result2.addReason(FlagReasonFactory::Unknown(), "more than 40 percent chambers are masked!");
+            return result2;
+          }
+          if (oldSector == currentSector) { // if chambers are masked within same sector
+            nMaskedHChInASec++;
+            if (nMaskedHChInASec == NCHAMBERPERSEC * 2) { // if entire sector is masked
+              result2 = Quality::Bad;
+              result2.addReason(FlagReasonFactory::Unknown(), "some sectors are entirely masked");
+              return result2;
+            }
+          } else {
+            nMaskedHChInASec = 1;
+          }
+        } // endif masked chamber
+        oldSector = currentSector;
+      } // hcid loop ends
+
+      overAllQuality = Quality::Good;
+      if (result1 == Quality::Good && result2 == Quality::Good) {
+        overAllQuality = Quality::Good;
+        overAllQuality.addReason(FlagReasonFactory::Unknown(), "All unmasked chambers working well!");
+        return overAllQuality;
+      }
+    }
+  }
+  return overAllQuality;
+  // return result1;
+}
+
+std::string CheckOnHc2d::getAcceptedType() { return "TH2"; }
+
+void CheckOnHc2d::beautify(std::shared_ptr<MonitorObject> mo, Quality checkResult)
+{
+  if (mo->getName() == "trackletsperHC2D") {
+    auto* h = dynamic_cast<TH2F*>(mo->getObject());
+
+    if (checkResult == Quality::Good) {
+      h->SetFillColor(kGreen);
```

Comment:
Does one see it if the histogram fill color for a TH2 changes if we draw it with COLZ option? I think we need a text box here